src/daemon/mod.rs

```diff
@@ -146,7 +146,15 @@ pub fn setup_panic_hook() {
         let bt = backtrace::Backtrace::new();
         if let Some(s) = panic_info.payload().downcast_ref::<&str>() {
             log::error!(
-                "panic occurred at line {} of file {}: {:?}\n{:?}",
+                "panic occurred at line {} of file {}: {}\n{:?}",
+                line,
+                file,
+                s,
+                bt
+            );
+        } else if let Some(s) = panic_info.payload().downcast_ref::<String>() {
+            log::error!(
+                "panic occurred at line {} of file {}: {}\n{:?}",
```

Comment:
nit: couldn't you use smth like `unwrap_or_else()` rather than duplicating the log message?